src/explore-education-statistics-admin/src/modules/chart-builder/ChartConfiguration.tsx

```diff
@@ -29,11 +36,128 @@ export interface ChartOptions {
   geographicId?: string;
 }
 
+interface InfographicChartOptionsProps {
+  releaseId: string;
+  fileId?: string;
+  onChange: (fileId: string) => void;
+}
+
+const InfographicChartOptions = ({
+  releaseId,
+  fileId,
+  onChange,
+}: InfographicChartOptionsProps) => {
+  const [chartFileOptions, setChartFileOptions] = React.useState<
+    SelectOption[]
+  >([]);
+
+  const [currentFileId, setCurrentFileId] = React.useState(fileId);
+
+  const [uploadName, setUploadName] = React.useState<string>('');
+
+  const [uploadFile, setUploadFile] = React.useState<File>();
+
+  const [uploading, setUploading] = React.useState<boolean>(false);
+
+  const loadChartFilesAsync = () => {
+    service.getChartFiles(releaseId).then(chartFiles => {
+      setChartFileOptions([
+        {
+          label: 'Upload a new file',
+          value: '',
+        },
+        ...chartFiles.map(({ title, filename }) => ({
+          label: title,
+          value: filename,
+        })),
+      ]);
+    });
+  };
+
+  const doUpload = () => {
+    if (uploadFile) {
+      setUploading(true);
+      service
+        .uploadChartFile(releaseId, {
+          name: uploadName,
+          file: uploadFile,
+        })
+        .then(() => loadChartFilesAsync())
+        .then(() => {
+          onChange(uploadFile.name);
+          setCurrentFileId(uploadFile.name);
```

Comment:
Is this line correct?  It's setting an id but using a name.

Reply:
FYI @negatron99

Reply:
It has nothing but the name to go on?